DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -128,7 +156,18 @@ class DMVisitorExpression : DMASTVisitor {
         }
 
         public void VisitCallableProcIdentifier(DMASTCallableProcIdentifier procIdentifier) {
-            Result = new Expressions.Proc(procIdentifier.Location, procIdentifier.Identifier);
+            if (_scopeMode == "static")
+            {
+                if (!DMObjectTree.GetDMObject(DreamPath.Root).HasProc(procIdentifier.Identifier))
+                {
+                    throw new CompileErrorException(new CompilerError(procIdentifier.Location, $"Global proc {procIdentifier.Identifier} does not exist"));
+                }
+                // TODO: substitute this to global.proc()
```

Comment:
This needs to either set `Result` or throw an exception so error handling ignores the expression.

Reply:
Apparently this was a bug that wasn't being caught in the test case. Should be fixed.